src/systems/systems/src/integrated_modular_avionics/core_processing_input_output_module.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{
+    shared::{ElectricalBusType, ElectricalBuses},
+    simulation::{
+        InitContext, Read, SimulationElement, SimulatorReader, SimulatorWriter, VariableIdentifier,
+        Write,
+    },
+};
+
+pub struct CoreProcessingInputOutputModule {
+    power_supply: ElectricalBusType,
+    is_powered: bool,
+    available_id: VariableIdentifier,
+    failure_indication_id: VariableIdentifier,
+    failure_indication: bool,
+}
+
+impl CoreProcessingInputOutputModule {
+    pub fn new(context: &mut InitContext, name: &str, power_supply: ElectricalBusType) -> Self {
+        Self {
+            power_supply,
+            is_powered: false,
+            available_id: context.get_identifier(format!("CPIOM_{}_AVAIL", name)),
+            failure_indication_id: context.get_identifier(format!("CPIOM_{}_FAILURE", name)),
+            failure_indication: false,
+        }
+    }
+
+    pub fn is_available(&self) -> bool {
+        self.is_powered & !self.failure_indication
+    }
+}
+
+impl SimulationElement for CoreProcessingInputOutputModule {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        let failure_value: f64 = reader.read(&self.failure_indication_id);
+        self.failure_indication = failure_value != 0.0;
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        if self.is_available() {
+            writer.write(&self.available_id, 1.0);
+        } else {
+            writer.write(&self.available_id, 0.0);
+        }
```

Comment:
possibility of simple assignment or tenery to save the use of if else? (not that familiar with rs sorry)

Reply:
I would actually recommend doing it like this:
```suggestion
        writer.write(&self.available_id, (self.is_available() as u64) as f64);
```

Reply:
actually this might be possible 

```suggestion
      writer.write(&self.available_id, self.is_available());
```

Reply:
fixed

Reply:
fixed